### PR TITLE
Scraper Optimization Bugfix

### PR DESCRIPTION
**Pull request by gshigin**
- Bugfix
- Tests for specific corner case

### CODE REVIEW

**Review by vporoshok — APPROVED** [review]

**Review by cherep58 — APPROVED** [review]

File: pp/wal/hashdex/scraper/encoding.h [hunk 1]
```diff
@@ -301,20 +301,23 @@ class LabelCodec {
 
   static PROMPP_ALWAYS_INLINE uint32_t read_val_partial(const char*& p, uint8_t sz) noexcept {
     if (sz == 0b01) [[likely]] {
-      return *p++;
+      uint8_t v;
```
Comment (cherep58) [inline pp/wal/hashdex/scraper/encoding.h]: this code block duplicated in code below. It seems like an template method
